src/main.ts

```diff
@@ -278,6 +279,7 @@ export function activate(context: vscode.ExtensionContext): ReturnType<typeof ge
 export class Extension {
     readonly extensionRoot: string
     readonly logger: Logger
+    readonly eventBus = new EventBus()
```

Comment:
For purely cosmetic reasons, I would have put the declaration here and the definition in `constructor()`

Reply:
We had better call `new EventBus()` here, which assures that `eventBus` already exists.

Reply:
You can see at `out/src/main.js`:

```js
class Extension {
    constructor() {
        this.eventBus = new eventbus_1.EventBus();
        this.extensionRoot = path.resolve(`${__dirname}/../../`);
        // We must create an instance of Logger first to enable
        // adding log messages during initialization.
        this.logger = new logger_1.Logger();
        this.addLogFundamentals();
```